.github/workflows/bot.yml

```diff
@@ -16,18 +16,36 @@ jobs:
     strategy:
       matrix:
         include:
-          - scala: "scala-2.11"
-            spark: "spark2"
-          - scala: "scala-2.11"
-            spark: "spark2,spark-shade-unbundle-avro"
-          - scala: "scala-2.12"
-            spark: "spark3.1.x"
-          - scala: "scala-2.12"
-            spark: "spark3.1.x,spark-shade-unbundle-avro"
-          - scala: "scala-2.12"
-            spark: "spark3"
-          - scala: "scala-2.12"
-            spark: "spark3,spark-shade-unbundle-avro"
+          - scalaProfile: "scala-2.11"
+            sparkProfile: "spark2"
+            sparkVersion: "2.4.4"
+
+          # Spark 3.1.x
+          - scalaProfile: "scala-2.12"
+            sparkProfile: "spark3.1.x"
+            sparkVersion: "3.1.0"
+
+          - scalaProfile: "scala-2.12"
+            sparkProfile: "spark3.1.x"
+            sparkVersion: "3.1.1"
+
+          - scalaProfile: "scala-2.12"
+            sparkProfile: "spark3.1.x"
+            sparkVersion: "3.1.2"
+
+          - scalaProfile: "scala-2.12"
+            sparkProfile: "spark3.1.x"
+            sparkVersion: "3.1.3"
+
+          # Spark 3.2.x
+          - scalaProfile: "scala-2.12"
+            sparkProfile: "spark3"
+            sparkVersion: "3.2.0"
+
+          - scalaProfile: "scala-2.12"
```

Comment:
Would this be okay with gh action minutes? @xushiyan

Reply:
> Would this be okay with gh action minutes? @xushiyan

Ci is performed for about 5-6 minutes.

Reply:
@vinothchandar gh actions only run mvn install atm; in https://github.com/apache/hudi/pull/5082 we adding some basic testcases covering quickstart for different spark versions. CI limit-wise we're good.